cloudmailer/cloudmailer.py

```diff
@@ -255,31 +255,77 @@ def listFile(textfile):
     lfile.close()
     return items
 
+# getServergroupsAndVms
+#
+# The function creates two maps, nodes and servergroups
+# For each node, the map nodes describes the list of server groups which have at least one server hosted in the node.
+# For each server group, the map servergroups describes the list of nodes which host at least one server of the server group.
+#
+# input
+# data: information about the Openstack platform retrieved from API call
+# nodelist: list of nodes we obtained in input
+#
+# output
+# nodes: dict containing [node->list of server groups] mappings
+# servergroups: dict containing [server group->list of nodes] mappings
 def getServergroupsAndVms(data,nodelist):
-
+    # create empty nodes map
```

Comment:
A friendly comment. There's a lot of excellent documentation here, in some cases it maybe a bit over the top though. In general documentation should add context you don't get from the code.  I'll try to add comments where I think the documentation may be too much, but this is mainly minor tuning and can be merged as is too.

So here,  
# create empty nodes map
may be superfluous as the next line does the exact thing the comment says